tdrs-frontend/src/components/ComboBox/ComboBox.jsx

```diff
@@ -1,21 +1,72 @@
-import React, { useEffect, useRef } from 'react'
+import React, { useEffect } from 'react'
 import PropTypes from 'prop-types'
 import comboBox from 'uswds/src/js/components/combo-box'
 
-const ComboBox = ({ children }) => {
-  const selectRef = useRef()
+/**
+ * @param {Component(s)} children - One or more React components to be rendered
+ * as options in the STT combo box.
+ * @param {function} handleSelect - A function to set the state of the STT
+ * in EditProfile upon selection of an STT.
+ * @param {string} selected - The name value of the STT. To be used as the
+ * value of the select component.
+ * @param {function} handleBlur - A function to run validation from
+ * EditProfile on the Combo Box. Runs on blur of combo box element.
+ * @param {string} error - If validation in EditProfile component throws
+ * an error then it is passed to combo box to render the error information.
+ * @param {string} name - A string used for the name and id values of
+ * the combo box.
+ * @param {string} placeholder - A string used as a placeholder
+ * in the combo box.
+ */
+const ComboBox = ({
+  children,
+  handleSelect,
+  selected,
+  handleBlur,
+  error,
+  name,
+  placeholder,
+}) => {
   useEffect(() => {
     // The combo box was not rendering as a combo box without this line
     comboBox.init()
-    // This solved the issue when tabbing through the form on EditProfile,
-    // a selection was automatically being made on the first option
-    selectRef.current.value = ''
+
+    const input = document.querySelector('.usa-combo-box__input')
+    if (input) {
+      if (error) {
+        input.classList.add('usa-combo-box__input--error')
+      }
+
+      if (!error) {
+        input.classList.remove('usa-combo-box__input--error')
+      }
+    }
   })
+
   return (
-    <label className="usa-label" htmlFor="sttList">
+    <label className="usa-label" htmlFor={name.toUpperCase()}>
       Associated State, Tribe, or Territory
-      <div className="usa-combo-box" data-placeholder="- Select or Search -">
-        <select className="usa-select" name="stt" id="stt" ref={selectRef}>
+      {error && (
+        <span
+          className="usa-error-message"
+          id="input-error-message"
```

Comment:
@hilvitzs This looks like another location where we may be introducing the same id for different elements, which we'll want to resolve.